http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpSamDAO.scala

```diff
@@ -48,6 +46,29 @@ class HttpSamDAO[F[_]](httpClient: Client[F],
 ) extends SamDAO[F]
     with Http4sClientDsl[F] {
   private val saScopes = Seq(PlusScopes.USERINFO_EMAIL, PlusScopes.USERINFO_PROFILE, StorageScopes.DEVSTORAGE_READ_ONLY)
+  private val leoSaTokenRef = Ref.ofEffect(getLeoAuthTokenInteral)
+
+  def registerLeo(implicit ev: Ask[F, TraceId]): F[Unit] =
+    for {
+      leoToken <- getLeoAuthToken
+      isRegistered <- httpClient.expectOr[RegisterInfoResponse](
+        Request[F](
+          method = Method.GET,
+          uri = config.samUri.withPath(Uri.Path.unsafeFromString(s"/register/user/v2/self/info")),
+          headers = Headers(leoToken)
+        )
+      )(onError)
+      _ <- httpClient
+        .successful(
+          Request[F](
+            method = Method.POST,
+            uri = config.samUri.withPath(Uri.Path.unsafeFromString(s"/register/user/v2/self")),
+            body = Stream.emits("app.terra.bio/#terms-of-service".getBytes(UTF_8)),
```

Comment:
what exactly does POSTing this payload do?

Reply:
it's just a required field for [this sam api](https://sam.dsde-dev.broadinstitute.org/#/Users/createUserV2)...it's always the same...not sure exactly what it does